src/MeetingsSDKAdapter.js

```diff
@@ -0,0 +1,172 @@
+   */
+  addLocalMedia(meetingID) {
+    const meeting = this.fetchMeeting(meetingID);
+
+    // default media setting
+    const mediaSettings = {
+      receiveVideo: true,
+      receiveAudio: true,
+      receiveShare: false,
+      sendVideo: true,
+      sendAudio: true,
+      sendShare: false,
+    };
+
+    meeting.getMediaStreams(mediaSettings).then(([localStream, localShare]) => {
+      meeting.addMedia({
+        localShare,
+        localStream,
+        mediaSettings,
+      });
+    });
+  }
+
+  /**
+   * Update the meeting object with media attached based on a given event type.
+   *
+   * @param {string} ID     ID of the meeting to fetch
+   * @param {Object} media  new media stream to attach to the meeting object based on a given event type
+   * @memberof MeetingsSDKAdapter
+   * @private
+   */
+  attachMedia(ID, {type, stream}) {
+    const meeting = this.meetings[ID];
+
+    switch (type) {
+      case MEDIA_TYPE_LOCAL:
+        this.meetings[ID] = {
+          ...meeting,
+          localAudio: stream.getAudioTracks()[0],
+          localVideo: stream.getVideoTracks()[0],
+        };
+        break;
+      default:
+        this.meetings[ID] = {...meeting};
+        break;
+    }
+  }
+
+  /**
+   * Creates and returns an observable to the new meeting object.
+   *
+   * @param {string} target destination to start the meeting with
+   * @returns {Observable.<Meeting>}
+   * @memberof MeetingsSDKAdapter
+   */
+  createMeeting(target) {
+    return from(this.datasource.meetings.create(target)).pipe(
+      map(({id, destination, sipuri}) => {
+        this.meetings[id] = {
+          ID: id,
+          title: sipuri || destination,
+          localVideo: null,
+          localAudio: null,
+          localShare: null,
+          remoteStream: null,
+          remoteShare: null,
+        };
+
+        return this.meetings[id];
+      })
+    );
+  }
+
+  /**
+   * Returns a SDK meeting object retrieved from the collection.
+   *
+   * @param {string} ID ID of the meeting to fetch.
+   * @returns {Meeting}
+   * @memberof MeetingsSDKAdapter
+   * @private
+   */
+  fetchMeeting(ID) {
+    return this.datasource.meetings.getMeetingByType('id', ID);
+  }
+
+  /**
+   * Returns an observable that emits meeting data of the given ID.
+   *
+   * @param {string} ID ID of meeting to get
+   * @returns {Observable.<Meeting>}
+   * @memberof MeetingsSDKAdapter
+   */
+  getMeeting(ID) {
+    if (!(ID in this.getMeetingObservables)) {
+      const meeting = this.fetchMeeting(ID);
+
+      const getMeeting$ = Observable.create((observer) => {
+        if (this.meetings[ID]) {
+          observer.next(this.meetings[ID]);
+        } else {
+          observer.error(new Error(`Could not find meeting with ID "${ID}"`));
+        }
+
+        observer.complete();
+      });
+
+      // Listen to attach mediaStream source objects to the existing meeting
+      const meetingWithReadyEvent$ = fromEvent(meeting, EVENT_MEDIA_READY).pipe(
```

Comment:
It took me a while to realize this was the sdk meeting object:

```suggestion
      const meetingWithReadyEvent$ = fromEvent(sdkMeeting, EVENT_MEDIA_READY).pipe(
```